### PR TITLE
feat(refactor): delete_contig_elements

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR fixes:

- See above

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: lib/MIP/Contigs.pm [hunk 1]
```diff
@@ -88,6 +89,60 @@ sub check_select_file_contigs {
...
+    };
+
+    check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
+
+    my %contig_to_remove = map { $_ => 1 } @{$remove_contigs_ref};
+
+    ### Make sure that contig is removed independent of genome source i.e prefix or not
+    ## If contigs has prefix
+    if ( defined $contigs_ref->[0]
+        && $contigs_ref->[0] =~ / ^chr /xsm )
```
Comment (jemten) [inline lib/MIP/Contigs.pm:130]:
```suggestion
        && $contigs_ref->[0] =~ / \A chr /xsm )
```

File: lib/MIP/Contigs.pm [hunk 2]
```diff
@@ -88,6 +89,60 @@ sub check_select_file_contigs {
...
+    {
+
+        ## And remove contigs has not
+        if ( defined $remove_contigs_ref->[0]
+            && $remove_contigs_ref->[0] !~ / ^chr /xsm )
```
Comment (jemten) [inline lib/MIP/Contigs.pm:135]:
```suggestion
            && $remove_contigs_ref->[0] !~ / \A chr /xsm )
```

File: lib/MIP/Contigs.pm [hunk 3]
```diff
@@ -88,6 +89,60 @@ sub check_select_file_contigs {
...
+        {
+
+            ## Add prefix to keys to match contigs prefix
+            %contig_to_remove = map { q{chr} . $_ => 1 } @{$remove_contigs_ref};
```
Comment (jemten) [inline lib/MIP/Contigs.pm:139]: Do we need a special case for the MT?

Comment (henrikstranneheim) [reply lib/MIP/Contigs.pm]: I added one

**Pull request merged**